src/main.rs

```diff
@@ -50,25 +55,28 @@ fn apply_gain(gain: &str) {
         .expect("Failed to update gain");
 }
 
+fn gain_changed(current: f64, new: f64) -> bool {
+    // Stringify to miniDSP format too see if the gain has changed
+    let current_gain_string = format!("{:.1}", current);
+    let new_gain_string = format!("{:.1}", new);
+    current_gain_string != new_gain_string
+}
+
 fn update_gain(current: f64, new: f64) -> bool {
     let gain_diff = new - current;
     let scaled_gain = current + (gain_diff * GAIN_SCALE);
     let final_gain = clamp(scaled_gain, MINIDSP_GAIN_MIN, MINIDSP_GAIN_MAX);
 
-    // Stringify to miniDSP format too see if the gain has changed
-    let old_gain_string = format!("{:.1}", current);
-    let new_gain_string = format!("{:.1}", final_gain);
-    let gain_changed = old_gain_string != new_gain_string;
-
-    if gain_changed {
-        apply_gain(&new_gain_string);
+    let changed = gain_changed(current, final_gain);
```

Comment:
`gain_changed` was a better name imo. By returning to that we can also remove the comment at output. 🥇